### PR TITLE
Add isolation2 and parallel-retrieve test matrix entries

### PR DESCRIPTION
**Pull request by edespino**
Adds two test matrix configurations:
- ic-isolation2 for isolation2 installcheck cases
- ic-parallel-retrieve-cursor for isolation2 parallel retrieve cursor tests

Default settings apply:
- num_primary_mirror_pairs: 3
- enable_cgroups: false
- enable_core_check: true
- pg_settings.optimizer: ""

I fully expect these new test cases to fail due to the need to update tests to account for the recent addition of `relisdynamic`.

### CODE REVIEW

**Review by my-ship-it — DISMISSED** [review]

**Comment by edespino** [comment]
@avamingli Can you assist with this?

**Comment by edespino** [comment]
The `isolation2` schedule is generating a core.

```
Thread 2 (Thread 0x7fb0301b2640 (LWP 11867)):
#0  0x00007fb03c6b4aff in poll () from /lib64/libc.so.6
#1  0x00007fb03adf9da9 in rxThreadFunc () from /usr/local/cloudberry-db-99.0.0/lib/postgresql/interconnect.so
#2  0x00007fb03c63cd22 in start_thread () from /lib64/libc.so.6
#3  0x00007fb03c6c1d40 in clone3 () from /lib64/libc.so.6

Thread 1 (Thread 0x7fb03b18bd00 (LWP 11865)):
#0  0x00007fb03c63ea6c in __pthread_kill_implementation () from /lib64/libc.so.6
#1  0x00007fb03c5f1686 in raise () from /lib64/libc.so.6
#2  0x00007fb03c5db833 in abort () from /lib64/libc.so.6
#3  0x00007fb03d2aa206 in errfinish () from /usr/local/cloudberry-db-99.0.0/lib/libpostgres.so
#4  0x00007fb03cceea2b in doNotifyingCommitPrepared.cold () from /usr/local/cloudberry-db-99.0.0/lib/libpostgres.so
#5  0x00007fb03d3548bf in notifyCommittedDtxTransaction () from /usr/local/cloudberry-db-99.0.0/lib/libpostgres.so
#6  0x00007fb03cdc70d3 in CommitTransaction () from /usr/local/cloudberry-db-99.0.0/lib/libpostgres.so
#7  0x00007fb03cdc842d in CommitTransactionCommand () from /usr/local/cloudberry-db-99.0.0/lib/libpostgres.so
#8  0x00007fb03d157378 in exec_simple_query () from /usr/local/cloudberry-db-99.0.0/lib/libpostgres.so
#9  0x00007fb03d15b6c9 in PostgresMain () from /usr/local/cloudberry-db-99.0.0/lib/libpostgres.so
#10 0x00007fb03d0b9ce0 in ServerLoop () from /usr/local/cloudberry-db-99.0.0/lib/libpostgres.so
#11 0x00007fb03d0bace3 in PostmasterMain () from /usr/local/cloudberry-db-99.0.0/lib/libpostgres.so
#12 0x00000000004027db in main ()
$1 = {si_signo = 6, si_errno = 0, si_code = -6, _sifields = {_pad = {11865, 1000, 0 <repeats 26 times>}, _kill = {si_pid = 11865, si_uid = 1000}, _timer = {si_tid = 11865, si_overrun = 1000, si_sigval = {sival_int = 0, sival_ptr = 0x0}}, _rt = {si_pid = 11865, si_uid = 1000, si_sigval = {sival_int = 0, sival_ptr = 0x0}}, _sigchld = {si_pid = 11865, si_uid = 1000, si_status = 0, si_utime = 0, si_stime = 0}, _sigfault = {si_addr = 0x3e800002e59, _addr_lsb = 0, _addr_bnd = {_lower = 0x0, _upper = 0x0}}, _sigpoll = {si_band = 4294967307865, si_fd = 0}, _sigsys = {_call_addr = 0x3e800002e59, _syscall = 0, _arch = 0}}}
[2024-12-19 03:38:11] GDB analysis completed successfully
```

File: src/test/isolation2/output/parallel_retrieve_cursor/explain.source [hunk 1]
```diff
@@ -115,48 +115,52 @@ VACUUM ANALYZE pg_class;
...
+ Seq Scan on pg_catalog.pg_class                                                                                                                                                                                                                                                                                                                                                                                                                                  
+   Output: oid, relname, relnamespace, reltype, reloftype, relowner, relam, relfilenode, reltablespace, relpages, reltuples, relallvisible, reltoastrelid, relhasindex, relisshared, relpersistence, relkind, relnatts, relchecks, relhasrules, relhastriggers, relhassubclass, relrowsecurity, relforcerowsecurity, relispopulated, relreplident, relispartition, relisivm, relisdynamic, relrewrite, relfrozenxid, relminmxid, relacl, reloptions, relpartbound 
+ Settings: enable_incremental_sort = 'on'
```
Comment (avamingli) [inline src/test/isolation2/output/parallel_retrieve_cursor/explain.source:121]: Config `enable_incremental_sort = on` manually when run tests?  Seems we should not have this GUC changed.

Comment (edespino) [reply src/test/isolation2/output/parallel_retrieve_cursor/explain.source:121]: > Config `enable_incremental_sort = on` manually when run tests? Seems we should not have this GUC changed.

@avamingli From a test execution perspective, this was not intentionally changed by me. The explain.source file has this enabled by default. If this needs to be changed, it should be performed with another PR? yes/no

Comment (avamingli) [reply src/test/isolation2/output/parallel_retrieve_cursor/explain.source:121]: Ok, yes.

**Review by avamingli — APPROVED** [review]

**Review by my-ship-it — APPROVED** [review]

**Pull request merged**